api/v1/swagger.yaml

```diff
@@ -1174,6 +1174,561 @@ paths:
+    put:
+      security:
+        - jwt: []
+      tags: [ collectors ]
+      summary: Update the specified collector.
+      description: >-
+        Update the specified collector's config data. If a field is not included
+        in the querybody, that field will be set to null.
+      operationId: updateCollector
+      parameters:
+        -
+          name: key
+          in: path
+          description: The id or name of the collector to update.
+          required: true
+          type: string
+        -
+          name: queryBody
+          in: body
+          description: Request body.
+          required: true
+          schema:
+            type: object
+            description: Fields permitted for putting.
+            properties:
+              description:
+                type: string
+                maxLength: 4096
+                description: >
+                  Describes this Collector; helps a user determine which Collectors they should assign to run their collection.
+              helpEmail:
+                type: string
+                description: >
+                  The email address where a user can go to get more help about the collector.
+                maxLength: 254
+              helpUrl:
+                type: string
+                maxLength: 2082
+                description: >
+                  The url where a user can go to get more help about the collector.
+      responses:
+        403:
+          $ref: "#/responses/403"
+        404:
+          $ref: "#/responses/404"
+
+  # ---------------------------------------------------------------------------
+
+  /collectors/{key}/deregister:
+    post:
+      security:
+        - jwt: []
+      summary: Access restricted to Refocus Collector only.
+      tags: [ collectors ]
+      description: >-
+        Access restricted to Refocus Collector only. Only valid if the collector is
+        registered and its status is STOPPED.
+      operationId: deregisterCollector
+      parameters:
+        -
+          name: key
+          in: path
+          description: The id or name of the collector to retrieve.
+          required: true
+          type: string
+      responses:
+        200:
+          description: Success, returns the specified collector's metadata.
+          schema:
+            $ref: "#/definitions/CollectorResponse"
+        400:
+          $ref: "#/responses/400"
+        403:
+          $ref: "#/responses/403"
+        404:
+            $ref: "#/responses/404"
+
+  # ---------------------------------------------------------------------------
+
+  /collectors/{key}/heartbeat:
+    post:
+      security:
+        - jwt: []
+      summary: Access restricted to Refocus Collector only.
+      tags: [ collectors ]
+      description: >-
+        Access restricted to Refocus Collector only.
+        Sends logs to refocus. This request may only be issued by a RUNNING or PAUSED
+        Refocus Collector.
+      operationId: sendHeartbeat
```

Comment:
Since this is POST, using sendHeartbeat is kind of misleading. What do you think?